### DISCUSSION TITLE
should contain key: :discord/token

### DISCUSSION THREAD

**Discussion by VoidAnomaly** [activexray/Doplarr#128] [Q&A]
I'm getting an error with the discord token. I followed the instructions and copied and pasted my bot's token into the proper place in the config file. Not sure why it's not working. Am I supposed to use the application ID or public key instead of the bot token?

It looks like this, this isn't that actual token, just gibberish to show how it looks.

:discord/token "oaspofTg2MzMxNz123dfasfDM123noiinoIwNg.iuniga.xmkasdffICAL-7FYfn7teBdGsuasdfeVQZJko"

And then I get this when trying to run the .jar

-- Spec failed --------------------

  {}

should contain key: :discord/token

| key            | spec    |
|================+=========|
| :discord/token | string? |

**Comment by hello0fficer** [comment]
Did you ever figure this out? I'm getting the same error.

**Reply by VoidAnomaly** [reply]
No, I ended up switching to Requestrr, and then starting earlier this year I started using Overseer instead and have been using that ever since.

**Reply by activexray** [reply]
This would mean the config file isn't being parsed (make sure there is no preceding semicolon, as those are comments in EDN), or the `-Dconfig=<path to config file>` wasn't supplied.

**Reply by hello0fficer** [reply]
I did supply the path to the config. No semi preceding either. 



user1@hockeypuck:~$ java -jar /lib/doplarr/doplarr.jar -Dconfig=/lib/doplarr/config.edn
2024-08-15T23:45:54.913Z hockeypuck FATAL [doplarr.config:68] - Error in  configuration
-- Spec failed --------------------

  {}

should contain key: :discord/token

| key            | spec    |
|================+=========|
| :discord/token | string? |

-- Relevant specs -------

:doplarr.config.specs/config:
  (clojure.spec.alpha/and
   (clojure.spec.alpha/keys
    :req
    [:discord/token]
    :opt
    [:discord/max-results
     :discord/requested-msg-style
     :radarr/quality-profile
     :sonarr/quality-profile
     :sonarr/language-profile
     :sonarr/season-folders
     :overseerr/default-id
     :sonarr/rootfolder
     :radarr/rootfolder]
    :opt-un
    [:doplarr.config.specs/partial-seasons
     :doplarr.config.specs/log-level])
   :doplarr.config.specs/has-backend
   (doplarr.config.specs/matched-keys :sonarr/url :sonarr/api)
   (doplarr.config.specs/matched-keys :radarr/url :radarr/api)
   (doplarr.config.specs/matched-keys :overseerr/url :overseerr/api))

-------------------------
Detected 1 error
